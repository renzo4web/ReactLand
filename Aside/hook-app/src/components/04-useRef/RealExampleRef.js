import React, { useState } from "react";
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";

const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div className="bg-light p-3">
      <h4>Real Example Ref</h4>

      {show && <MultipleCustomHooks />}
      <button onClick={() => setShow(!show)}>{show ? "Hide" : "Show"}</button>
    </div>
  );
};

export default RealExampleRef;
