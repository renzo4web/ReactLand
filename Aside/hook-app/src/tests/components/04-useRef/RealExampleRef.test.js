import { shallow } from "enzyme";
import React from "react";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe("<RealExampleRef />", () => {
  test("should the component render properly", () => {
    const wrapper = shallow(<RealExampleRef />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should show the component <MultipleCustomHooks />", () => {
    const wrapper = shallow(<RealExampleRef />);

    wrapper.find("button").simulate("click");

    expect(
      wrapper.find("MultipleCustomHooks").exists()
    ).toBe(true);
  });
});
