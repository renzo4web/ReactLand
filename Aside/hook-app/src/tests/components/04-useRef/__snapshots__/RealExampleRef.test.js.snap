// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef /> should the component render properly 1`] = `
<div
  className="bg-light p-3"
>
  <h4>
    Real Example Ref
  </h4>
  <button
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
